src/types/mod.rs

```diff
@@ -38,6 +38,7 @@ pub enum PrimitiveType {
     CharType,                 // : *
     BooleanType,              // : *
     UnitType,                 // : *
+    Ptr,                      // : * -> *
```

Comment:
I saw in an earlier commit that `ref` in primitive had this comment. Is this correct?

Reply:
Yep! `* -> *` is the _kind_ of `Ptr`. In kind terminology `*` means type so `* -> *` means `Ptr` is a function which takes a type and returns a type, so you can give it `i32` as an argument: `Ptr i32` and you get back the type of a pointer to an integer.
Kinds aren't really checked at all during compilation yet, save for some asserts during codegen. We may want to check these earlier (during unification) sometime in the future for better error messages when users try something like `foo : Ptr` instead of `foo : Ptr i32` for example.